File: INTRODUCCION-POO/EJERCICIO-1/Persona.py
# INTRODUCCIÓN A LA PROGRAMACIÓN ORIENTADA A OBJETOS EN PYTHON
#Crea una clase Persona con nombre, edad y ciudad
#a) Agrega un método para mostrar el saludo: “Hola, soy {nombre} de {ciudad}”
#b) Crea tres personas y muestra su saludo
#c) Agrega un método para verificar si es mayor de edad

class Persona:
    
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre  
        self.edad = edad      
        self.ciudad = ciudad  

    
    def saludo(self):
        return f"Hola, soy {self.nombre} de {self.ciudad}"

    
    def es_mayor_de_edad(self):
        return self.edad >= 18  



persona1 = Persona("Juan", 25, "Madrid")
persona2 = Persona("Ana", 17, "Barcelona")
persona3 = Persona("Carlos", 30, "Valencia")

# Mostrar saludos
print(persona1.saludo()) 
print(persona2.saludo())  
print(persona3.saludo())  

# Verificar si son mayores de edad
print(f"{persona1.nombre} es mayor de edad: {persona1.es_mayor_de_edad()}")
print(f"{persona2.nombre} es mayor de edad: {persona2.es_mayor_de_edad()}")
print(f"{persona3.nombre} es mayor de edad: {persona3.es_mayor_de_edad()}")

File: INTRODUCCION-POO/EJERCICIO-1/Persona.java
// INTRODUCCIÓN A LA PROGRAMACIÓN ORIENTADA A OBJETOS EN JAVA


class Persona {
    // Atributos de la clase
    String nombre;
    int edad;
    String ciudad;

    // Constructor de la clase
    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    // Método para mostrar un saludo
    public String saludo() {
        return "Hola, soy " + this.nombre + " de " + this.ciudad;
    }

    // Método para verificar si es mayor de edad
    public boolean esMayorDeEdad() {
        return this.edad >= 18;  // Retorna true si es mayor de edad, false si no
    }

    // Método principal (main) para probar la clase
    public static void main(String[] args) {
        // Crear tres objetos de la clase Persona
        Persona persona1 = new Persona("Juan", 25, "Madrid");
        Persona persona2 = new Persona("Ana", 17, "Barcelona");
        Persona persona3 = new Persona("Carlos", 30, "Valencia");

        // Mostrar saludos
        System.out.println(persona1.saludo());  // Saludo de persona1
        System.out.println(persona2.saludo());  // Saludo de persona2
        System.out.println(persona3.saludo());  // Saludo de persona3

        // Verificar si son mayores de edad
        System.out.println(persona1.nombre + " es mayor de edad: " + persona1.esMayorDeEdad());
        System.out.println(persona2.nombre + " es mayor de edad: " + persona2.esMayorDeEdad());
        System.out.println(persona3.nombre + " es mayor de edad: " + persona3.esMayorDeEdad());
    }
}